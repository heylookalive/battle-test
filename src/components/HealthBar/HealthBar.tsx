import React from 'react';
import styled from 'styled-components';

type HealthBarProps = {
  value: number;
};

const HealthBarWrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr 30px;
  gap: 10px 10px;
  grid-template-areas:
    '.'
    '.';
`;

const BarWrapper = styled.div`
  position: relative;
  background-color: #f0f0f0;
  border-radius: 2px;
`;

const Bar = styled.div<{ value: number }>`
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: ${(props) => props.value}%;
  border-radius: 2px;
  background-color: limegreen;
`;

const Label = styled.div`
  text-align: center;
`;

export default function HealthBar({ value }: HealthBarProps) {
  const noNegativeInts = value < 0 ? 0 : value;
  return (
    <HealthBarWrapper>
      <BarWrapper>
        <Bar value={value} />
      </BarWrapper>
      <Label data-testid="healthbar-label">{noNegativeInts}</Label>
    </HealthBarWrapper>
  );
}
