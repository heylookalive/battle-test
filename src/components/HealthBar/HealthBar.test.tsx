import { render } from '@testing-library/react';
import React from 'react';
import HealthBar from './HealthBar';

describe('<HealthBar />', () => {
  it('renders as expected', () => {
    const { container } = render(<HealthBar value={73} />);
    expect(container).toMatchSnapshot();
  });

  it('renders as expected, given a negative value', () => {
    const { getByTestId } = render(<HealthBar value={-10} />);
    expect(getByTestId('healthbar-label')).toHaveTextContent('0');
  });
});
