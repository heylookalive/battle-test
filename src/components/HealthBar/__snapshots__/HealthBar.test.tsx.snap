// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HealthBar /> renders as expected 1`] = `
<div>
  <div
    class="sc-AxjAm ikqlAb"
  >
    <div
      class="sc-AxirZ dQUJrW"
    >
      <div
        class="sc-AxiKw ihDPnh"
        value="73"
      />
    </div>
    <div
      class="sc-AxhCb cinyOp"
      data-testid="healthbar-label"
    >
      73
    </div>
  </div>
</div>
`;
